test/text/lrc_text_parser_unit.js

```diff
@@ -0,0 +1,85 @@
+/*! @license
+ * Shaka Player
+ * Copyright 2016 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+goog.require('shaka.text.LrcTextParser');
+goog.require('shaka.util.BufferUtils');
+goog.require('shaka.util.StringUtils');
+
+describe('LrcTextParser', () => {
+  it('supports no cues', () => {
+    verifyHelper([],
+        '',
+        {periodStart: 0, segmentStart: 0, segmentEnd: 0});
+  });
+
+  it('handles a blank line at the start of the file', () => {
+    verifyHelper(
+        [
+          {startTime: 0, endTime: 2, payload: 'Test'},
+        ],
+        '\n\n' +
+        '[00:00.00]Test',
+        {periodStart: 0, segmentStart: 0, segmentEnd: 0});
+  });
+
+  it('handles a blank line at the end of the file', () => {
+    verifyHelper(
+        [
+          {startTime: 0, endTime: 2, payload: 'Test'},
+        ],
+        '[00:00.00]Test' +
+        '\n\n',
+        {periodStart: 0, segmentStart: 0, segmentEnd: 0});
+  });
+
+  it('handles no blank line at the end of the file', () => {
+    verifyHelper(
+        [
+          {startTime: 0, endTime: 2, payload: 'Test'},
+        ],
+        '[00:00.00]Test',
+        {periodStart: 0, segmentStart: 0, segmentEnd: 0,
+        });
+  });
+
+  it('supports multiple cues', () => {
+    verifyHelper(
+        [
+          {startTime: 0, endTime: 10, payload: 'Test'},
+          {startTime: 10, endTime: 20, payload: 'Test2'},
+          {startTime: 20, endTime: 22, payload: 'Test3'},
+        ],
+        '[00:00.00]Test\n' +
+        '[00:10.00]Test2\n' +
+        '[00:20.00]Test3',
+        {periodStart: 0, segmentStart: 0, segmentEnd: 0});
+  });
+
```

Comment:
Please add tests that include fractional seconds in 1-3 digits as your regex allows.

Reply:
Done!